packages/examples/cvat/exchange-oracle/src/handlers/completed_escrows.py

```diff
@@ -30,382 +27,259 @@
     postprocess_annotations,
     prepare_annotation_metafile,
 )
+from src.models.cvat import Job, Project
 from src.services.cloud.types import BucketAccessInfo
 from src.utils.assignments import parse_manifest
-from src.utils.logging import NullLogger
 
 
-class _CompletedEscrowsHandler:
```

Comment:
What was the point of removing the class here? Now you need to pass some shared variables into every function (e.g. logger).

Reply:
As a side note.

I think we would benefit here from adopting queues and workers approach and handle. This would abstract away retries logic and will allow us to focus just on task implementation at hand. 

State trackers then won't be doing any work and will just schedule jobs.

Reply:
The problem with a bunch of functions instead of a class is that at some point these functions become [this](https://github.com/cvat-ai/cvat/blob/12858584c8b33acb021becd239201de8440ed669/cvat/apps/engine/task.py#L556-L1516) - a single big function with lots of branching and context. And it can't be easily refactored or extended for another new use case.

Reply:
I’m afraid the example you provided doesn’t really connect with the class vs. functions discussion, nor with this particular case.

It shows instead how function (or method) can grow in complexity when lacking proper decomposition. Which is a completely valid, but separate issue.

Your argument instead implies that code written in functional languages must all suffer from having large, complex functions, which is simply not the case.

Another contradiction to your example is the fact that current implementation is actually less complex and more concise than previous class-based one.

So it's clearly not a question of whether to use class to reduce complexity, since the complexity itself arises due to different reasons.